lib/pubsub/topic.js

```diff
@@ -63,6 +63,27 @@ function Topic(pubsub, options) {
 }
 
 /**
+ * Format a message object as the upstream API expects it.
+ *
+ * @private
+ *
+ * @return {object}
+ */
+Topic.formatMessage_ = function(message) {
+  if (!message.data) {
+    message = {
+      data: new Buffer(message).toString('base64')
+    };
+  }
+
+  if (!util.is(message.data, 'string') && !util.is(message.data, 'buffer')) {
+    message.data = JSON.stringify(message.data);
```

Comment:
Yeah just keep publishRaw for the catch-all use case in the scenario that pubsub adds more optional parameters though I don't see why we need to support the case where they have a base64'd string they want to publish REALLY RAW.